### PR TITLE
Truncate ServiceAccount display names longer than 100 characters

### LINKED ISSUE #68: Rolesets with long names cannot be updated because SA display name exceeds 100 characters
We are autogenerating roleset names from other entity data and they can be >50 characters in length.

Vault generates a description for the service account that reads `Service account for Vault secrets backend role set <name>` and adds 51 characters of boilerplate. This overflows the 100 character length limit for the description and causes service account creation to fail.

### PR DESCRIPTION
**Pull request by mdgreenfield**
This will sha256(displayName[92:]) and concat the first 8
characters of the sha256 hash to the display name.

Fixes #68

### CODE REVIEW

File: plugin/role_set_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package gcpsecrets
+
+import (
+	"fmt"
+	"testing"
+)
+
+func Test_roleSetServiceAccountDisplayName(t *testing.T) {
```
Comment (agy) [inline plugin/role_set_test.go:8]: Style suggestion: I think that [table tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) would be cleaner and clearer here.

**Review by emilymye — DISMISSED** [review]

**Comment by mdgreenfield** [comment]
@emilymye is this something can get merged?

File: plugin/role_set.go [hunk 2]
```diff
@@ -415,6 +416,18 @@ func roleSetServiceAccountName(rsName string) (name string) {
 	return name
 }
 
+func roleSetServiceAccountDisplayName(name string) string {
+	hashLen := 8
+	fullDisplayName := fmt.Sprintf(serviceAccountDisplayNameTmpl, name)
+	displayName := fullDisplayName
+	if len(fullDisplayName) > serviceAccountDisplayNameMaxLen {
+		truncIndex := serviceAccountDisplayNameMaxLen - hashLen
+		h := fmt.Sprintf("%x", sha256.Sum256([]byte(fullDisplayName[truncIndex:])))
```
Comment (calvn) [inline plugin/role_set.go:425]: Should the sha256sum be calculated off the full `fullDisplayName` to reduce the chances of a collision or is there a particular reason we're only considering bytes within `fullDisplayName[truncIndex:]`? The final length on the display name seems to be truncated by `h[:hashLen]` anyways.

File: plugin/role_set.go [hunk 3]
```diff
@@ -415,6 +416,18 @@ func roleSetServiceAccountName(rsName string) (name string) {
 	return name
 }
 
+func roleSetServiceAccountDisplayName(name string) string {
+	hashLen := 8
```
Comment (calvn) [inline plugin/role_set.go:420]: Can we move this into the const declarations up top?

File: plugin/role_set.go (hunk 2 above)
Comment (calvn) [reply plugin/role_set.go:425]: Actually, the first part of the display name is constructed via `fullDisplayName[:truncIndex]` so even if the hash ends up being the same as another one that had the same suffix string, the first part would be unique if the full display name is different from the one that it "collided" with so I think it's fine.

**Review by calvn — DISMISSED** [review]

File: plugin/role_set.go (hunk 3 above)
Comment (mdgreenfield) [reply plugin/role_set.go]: fixed

**Review by calvn — APPROVED** [review]

**Pull request merged**